### DISCUSSION TITLE
Recommended approach for VSCode/Atom extensions

### DISCUSSION THREAD

**Discussion by jakobrosenberg** [serialport/node-serialport#2350] [Q&A]
VSCode extensions are installed in Node, but executed in Electron. This causes the wrong bindings to be downloaded.

I've worked around this by creating a utility that removes bindings at runtime and then redownloads the correct ones. If none are found, it then falls back to building the bindings on the host machine.

Is there a better way to get around this? Could it be solved if `node-serialport` used `prebuildify` to create its bindings?

@reconbot, I'll be happy to contribute in anyway I can to help solve this. I saw quite a few issues and usecases like mine, so I'm hoping it's a worthwhile effort.

**Comment by GazHank** [comment]
I guess this is another thing that the N-API migration should resolve...
